### PR TITLE
Refactor analyzer BindingTypeCodeRefactoringProviderTests

### LINKED ISSUE #1746: Reenable analyzer unit tests & release analyzer package
Tests were disabled as part of a move from feature/sdk-bindings to main as they depend on new worker extension releases. This issue is to re-enable these tests and update all necessary package version references once new packages are available

- [x] Reenable BindingTypeCodeRefactoringProviderTests
- [ ] Release stable analyzer package

### PR DESCRIPTION
**Pull request by liliankasem**
### Issue describing the changes in this PR

resolves #1746

Rewrite BindingTypeCodeRefactoringProvider tests to use different extensions & added case that doesn't have any extension dependencies as well. Fixed edge case with BinaryData

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: sdk/Sdk.Analyzers/Extensions/TypeSymbolExtensions.cs [hunk 1]
```diff
@@ -58,6 +58,11 @@ internal static string GetMinimalDisplayName(this ITypeSymbol type, SemanticMode
                 name = Regex.Match(name, @"IEnumerable<[^>]+>").Value;
             }
 
+            if (name.Contains("System."))
```
Comment (surgupta-msft) [inline sdk/Sdk.Analyzers/Extensions/TypeSymbolExtensions.cs:61]: double checking that we want `Contains()` here and not `Startswith("System.")`

**Review by surgupta-msft — APPROVED** [review]

File: sdk/Sdk.Analyzers/Extensions/TypeSymbolExtensions.cs (hunk 1 above)
Comment (liliankasem) [reply sdk/Sdk.Analyzers/Extensions/TypeSymbolExtensions.cs:61]: Either work I think but I like the idea of Startswith
